### PR TITLE
Fix issue when loading tenant wise email publishers

### PR DESCRIPTION
**Pull request by GANGANI**
Resolves: https://github.com/wso2/product-is/issues/7894

### CODE REVIEW

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/ConfigurationDAO.java [hunk 1]
```diff
@@ -223,12 +223,13 @@ void addFile(String fileId, String resourceId, String fileName, InputStream file
...
-     * Get files for the {@link ResourceType}.
+     * Get files for the resource by resource type Id and tenant ID.
      *
-     * @param resourceTypeId Id of the {@link ResourceType}.
+     * @param resourceTypeId Id of the {@link ResourceType}
+     * @param tenantID       Id of the tenant
      * @return A list of {@link ResourceFile} for the given resource.
      */
-    List<ResourceFile> getFilesByResourceType(String resourceTypeId) throws ConfigurationManagementException;
+    List<ResourceFile> getFilesByResourceType(String resourceTypeId, int tenantID) throws ConfigurationManagementException;
```
Comment (tharindu-bandara) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/ConfigurationDAO.java]:
```suggestion
    List<ResourceFile> getFilesByResourceType(String resourceTypeId, int tenantId) throws ConfigurationManagementException;
```

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/ConfigurationDAO.java [hunk 2]
```diff
@@ -223,12 +223,13 @@ void addFile(String fileId, String resourceId, String fileName, InputStream file
     List<ResourceFile> getFiles(String resourceId, String resourceTypeName, String resourceName) throws ConfigurationManagementException;
 
     /**
-     * Get files for the {@link ResourceType}.
+     * Get files for the resource by resource type Id and tenant ID.
```
Comment (tharindu-bandara) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/ConfigurationDAO.java]:
```suggestion
     * Get files for the resource by resource type Id and tenant id.
```

File: components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManagerImpl.java [hunk 3]
```diff
@@ -778,6 +778,25 @@ public ResourceFile addFile(String resourceTypeName, String resourceName, String
...
+    @Override
+    public List<ResourceFile> getFiles(String resourceTypeName, int tenantId) throws ConfigurationManagementException {
+
+        checkFeatureStatus();
+        validateRequest(resourceTypeName);
+        String resourceTypeId = getResourceTypeId(resourceTypeName);
+        List<ResourceFile> resourceFiles = getConfigurationDAO().getFilesByResourceType(resourceTypeId, tenantId);
+        if (CollectionUtils.isEmpty(resourceFiles)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Resource type: " + resourceTypeName + " does not have any files.");
```
Comment (mefarazath) [inline components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManagerImpl.java:790]: Is the tenant tracked in some other way (part of the log line)?
If not, would be better to included the tenantId or tenantDomain as part of the log

Comment (GANGANI) [reply components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/ConfigurationManagerImpl.java:790]: Fixed https://github.com/wso2/carbon-identity-framework/pull/3104/commits/58591c8a79d9819dc4e42f6c1697f7e6061b1682

**Pull request merged**